### ISSUE TITLE
Unable to integrate with Openstack Cinder: "libstorage: error authenticating"

### ISSUE THREAD

**Issue by pwojtyra** [rexray/rexray#1261]
# Summary
Unable to integrate RexRay with Openstack Cinder (HPE Helion Openstack). Rexray cannot authenticate. 
The error: libstorage: error authenticating
Since my Openstack installation is using self-signed certificates, I was trying to disable cert checking, but haven't found any documentation on that. 
 
## Version

REX-Ray
-------
SemVer: 0.11.3
OsArch: Linux-x86_64

## Expected Behavior
Rexray is connecting to the cinder service and is able to do actions (list,create,delete volumes)
Rexray is able to start as a service

## Actual Behavior
Rexray command throws error: libstorage: error authenticating
Rexray is not starting as a service. 

## Steps To Reproduce
Install rexray curl -sSL https://rexray.io/install | sh
Create config.yml like below
Run the command: rexray -l debug volume ls

## Configuration Files
/etc/rexray/config.yml
rexray:
  loglevel: debug
libstorage:
  logging:
    level: debug
    httpRequests: true
    httpResponses: true
  service: cinder
  client:
    tls: insecure
  server:
    services:
      cinder:
        driver: cinder
        cinder:
          authURL: https://*.*.*.*:5000/v3
          password: *******
          tenantName: *******
          identityApiVersion: 3
          userName: *******
          availabilityZoneName: nova
          domainName: Default
          tls: insecure

### Client Log
INFO[0000] configured default CSI module                 CSI_ENDPOINT=/run/docker/plugins/rexray.sock time=1535547553588
INFO[0000] DockerLegacyMode                              enabled=true time=1535547553589
DEBU[0000] trapped signals                               time=1535547553589
DEBU[0000] loading config                                buildType=client+agent+controller configFileName=rexray globalConfigFileExists=false globalConfigFilePath=/etc/rexray/rexray.yml ignoreExists=false time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/rexray.yml
DEBU[0000] cannot find global or user file               buildType=client+agent+controller configFileName=rexray globalConfigFileExists=false globalConfigFilePath=/etc/rexray/rexray.yml ignoreExists=false time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/rexray.yml
DEBU[0000] loading config                                buildType=client+agent+controller configFileName=config globalConfigFileExists=true globalConfigFilePath=/etc/rexray/config.yml ignoreExists=true time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/config.yml
DEBU[0000] disabled config file exist check              buildType=client+agent+controller configFileName=config globalConfigFileExists=true globalConfigFilePath=/etc/rexray/config.yml ignoreExists=true time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/config.yml
DEBU[0000] validating global config                      buildType=client+agent+controller configFileName=config globalConfigFileExists=true globalConfigFilePath=/etc/rexray/config.yml ignoreExists=true time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/config.yml
DEBU[0000] created new config                            buildType=client+agent+controller configFileName=config globalConfigFileExists=true globalConfigFilePath=/etc/rexray/config.yml ignoreExists=true time=1535547553589 userConfigFileExists=false userConfigFilePath=/root/.rexray/config.yml
DEBU[0000] initializing configuration
DEBU[0000] loading yaml for Global
DEBU[0000] loading yaml for CSI
DEBU[0000] loading global config file                    path=/etc/rexray/config.yml
DEBU[0000] usage template path                           path=/root/.rexray/usage.template time=1535547553591
DEBU[0000] loaded usage template                         source=UsageTemplate time=1535547553591
DEBU[0000] created scoped scope                          new=rexray.cli parentScopes=rexray,
DEBU[0000] updating log level                            level=debug time=1535547553595
INFO[0000] updated log level                             logLevel=debug
DEBU[0000] os.args                                       time=1535547553599 val=[rexray volume ls -l debug]
DEBU[0000] activating libStorage                         cmd=ls time=1535547553599
DEBU[0000] host is empty; initiliazing default services  time=1535547553609
DEBU[0000] libStorage auto service mode disabled; services defined  time=1535547553611
DEBU[0000] starting embedded libStorage server           time=1535547553611
DEBU[0000] created scoped scope                          new=libstorage.server parentScopes=rexray.cli,rexray,
INFO[0000] configured logging                            libstorage.logging.httpRequests=true libstorage.logging.httpResponses=true libstorage.logging.level=debug libstorage.logging.stderr=/home/centos/.rexray/var/log/libstorage-error.log libstorage.logging.stdout=/home/centos/.rexray/var/log/libstorage-out.log server=aquamarine-diver-kp time=1535547553678
DEBU[0000] parsed server auth property                   libstorage.server.auth.disabled=false server=aquamarine-diver-kp time=1535547553697
DEBU[0000] parsed server auth property                   libstorage.server.auth.key= server=aquamarine-diver-kp time=1535547553710
DEBU[0000] parsed server auth property                   libstorage.server.auth.alg=HS256 server=aquamarine-diver-kp time=1535547553717
DEBU[0000] parsed server auth property                   libstorage.server.auth.allow=[] server=aquamarine-diver-kp time=1535547553729
DEBU[0000] parsed server auth property                   libstorage.server.auth.deny=[] server=aquamarine-diver-kp time=1535547553741
INFO[0000] configured global auth                        libstorage.server.auth.alg=HS256 libstorage.server.auth.allow=[] libstorage.server.auth.deny=[] libstorage.server.auth.disabled=false libstorage.server.auth.key= server=aquamarine-diver-kp time=1535547553741
INFO[0000] initializing server                           server=aquamarine-diver-kp time=1535547553741
INFO[0000] initializing server services                  server=aquamarine-diver-kp time=1535547553741
DEBU[0000] configured result schema validation           enabled=false server=aquamarine-diver-kp time=1535547553745
DEBU[0000] got services map                              count=1 server=aquamarine-diver-kp time=1535547553749
DEBU[0000] processing service config                     server=aquamarine-diver-kp service=cinder time=1535547553749
DEBU[0000] getting scoped config for service             scope=libstorage.server.services.cinder server=aquamarine-diver-kp service=cinder time=1535547553749
DEBU[0000] created scoped scope                          new=libstorage.server.services.cinder parentScopes=libstorage.server,rexray.cli,rexray,
DEBU[0000] got driver name                               driverName=cinder server=aquamarine-diver-kp service=cinder time=1535547553762
Oops, an error occured!
 
  libstorage: error authenticating
 
To correct the error please review:
 
  - Debug output by using the flag -l debug
  - The REX-ray website at https://github.com/rexray/rexray
  - The on↓ine he↓p be↓ow

**Comment by clintkitson** [comment]
Hello @pwojtyra,

I believe this is related to https://github.com/rexray/rexray/issues/1190.

Looking for the community to help address this refactoring.

**Comment by pwojtyra** [comment]
Hi @clintkitson 

My openstack environment is using v2 and v1 cinder api  whereas issue #1113 refers to the scenario where cinder v1 api is not available.

I believe this issue is due to untrusted self-signed certs. Anybody has a working config.yml file with disabled cert checking wiht openstack? Or any instructions on how to add openstack untrusted cert to the raxray config to make it work? I appreciate any help.

**Comment by pwojtyra** [comment]
I've managed to work around this issue. As I expected it is due to certificates/TLS.

I've changed the authURL in config.yml to http address of one of my controllers - so tls was not used. Earlier I was using https to VIP of my controllers cluster. 
Since my Openstack installation returns storage api pointing on the https VIP I again hit the problem with untrusted certificate. This time it was resolved by adding the Openstack certificate file to /etc/pki/ca-trust/source/anchors/.
Strange is that this does not resolve the problem with initial authentication to the keystone api, so I need to stay with http single controller address in authURL.
I've seen that behavior on 4 of my nodes.

In other words (at least in my setup) is seems that during initial authentication to keystone trusted certificates from /etc/pki/ca-trust/source/anchors/ are not taken into account, whereas during cinder endpoint communication this is done.
